### PR TITLE
Date format

### PR DESCRIPTION
**Pull request by Sjmarf**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [x] This PR addresses one or more open issues that were assigned to me:
      - #33
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

Addresses #33 to make the date on posts and comments shorter. It now uses `s`, `m`, `h`, `d`, `w`, `mo`, `y`.

Also abstracted the timestamp to a `TimestampView` struct for easier reusability.

Screenshot 2023-07-06 at 22 38 49

### CODE REVIEW

**Comment by Sjmarf** [comment]
Closes #33

**Review by WestonHanners — APPROVED** [review]

File: Mlem/Logic/Time Parser.swift
Comment (Sjmarf) [reply Mlem/Logic/Time Parser.swift:15]: Theoretically it could. I can modify the code to display 'now' instead for values <1s?

Comment (mormaer) [reply Mlem/Logic/Time Parser.swift:15]: I'd prefer to see  it render _now_ over `0s` personally, does anyone else have any thoughts?

Comment (Sjmarf) [reply Mlem/Logic/Time Parser.swift:15]: It would definitely be better. I suppose it's likely to happen when an Mlem user posts a comment. I'll change it to display 'now' 👍

Comment (EricBAndrews) [reply Mlem/Logic/Time Parser.swift:15]: I'm a fan of "now" as well--that + conflicts and we're good to merge

File: Mlem/Views/Shared/TimestampView.swift
Comment (EricBAndrews) [inline Mlem/Views/Shared/TimestampView.swift:16]: Spacing should use `AppConstants.iconToTextSpacing`

**Review by EricBAndrews — CHANGES_REQUESTED** [review]

File: Mlem/Views/Shared/TimestampView.swift
Comment (Sjmarf) [reply Mlem/Views/Shared/TimestampView.swift:16]: There is no such `AppConstants.iconToTextSpacing` on my branch. My implementation of `TimestampView` accepts a `spacing` argument - by passing a value as this argument, all instances of `TimestampView` are able to use the same spacing as they did before this PR. 

The values `PostInteractionBar.iconToTextSpacing` and `CommentInteractionBar.iconToTextSpacing` are passed as the `spacing` argument in the relevant `TimestampView` instances. Is that what you meant?

Comment (EricBAndrews) [reply Mlem/Views/Shared/TimestampView.swift:16]: It's on master, must have come in a recent PR. I'll approve, once the merge conflicts are fixed we're good to go

**Review by EricBAndrews — APPROVED** [review]

File: Mlem/Views/Shared/TimestampView.swift
Comment (Sjmarf) [reply Mlem/Views/Shared/TimestampView.swift:16]: Ah I see. I've changed it 👍

**Review by EricBAndrews — APPROVED** [review]

**Comment by EricBAndrews** [comment]
Merged, thanks for your contribution!

**Pull request merged**
